include/ut_test_runner.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2023 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+#ifndef UT_TEST_RUNNER_H
+#define UT_TEST_RUNNER_H
+
+#include <gtest/gtest.h>
+#include <functional>
+#include <string>
+
+class UTTestRunner
```

Comment:
why is this in the external inferface, what would the user require of it?

Reply:
This class has test run related functions.
User can create an object like below and call the associated run functions:

```
UTTestRunner testRunner(argc, argv);
 return testRunner.runTests();
```

Reply:
what has this got to-do with the way that we run, it's related to gtest and not the way we work today.

the testRunner is hidden in the UT_cunit.c & UT_gtest.cpp